src/imap_mag/cli/fetchScience.py

```diff
@@ -0,0 +1,74 @@
+"""Program to retrieve and process MAG CDF files."""
+
+import typing
+from pathlib import Path
+
+import pandas as pd
+import typing_extensions
+
+from .. import appUtils
+from ..client.sdcDataAccess import ISDCDataAccess
+
+
+class FetchScienceOptions(typing.TypedDict):
+    """Options for SOC interactions."""
+
+    level: str
+    start_date: str
+    end_date: str
+    output_dir: str
+
+
+class FetchScience:
+    """Manage SOC data."""
+
+    __descriptors: list[str]
+    __flavors: list[str]
+
+    __data_access: ISDCDataAccess
+
+    def __init__(
+        self,
+        data_access: ISDCDataAccess,
+        descriptors: list[str] = ["norm", "burst"],
+        flavors: list[str] = ["-magi", "-mago"],
```

Comment:
Switch these magic strings for strongly typed enums?

Reply:
Good idea! Will do